src/renderer/api/websocket-api.ts

```diff
@@ -181,7 +181,7 @@ export class WebSocketApi<Events extends WebSocketEvents> extends (EventEmitter
     if (error) {
       const { reconnectDelay } = this.params;
 
-      if (reconnectDelay) {
+      if (reconnectDelay && this.socket) {
```

Comment:
Shouldn't we change the type to be something like `protected socket?: WebSocket | null;` to catch these errors with the compiler?